cpp/src/arrow/util/value_parsing.h

```diff
@@ -273,6 +273,36 @@ inline bool ParseUnsigned(const char* s, size_t length, uint64_t* out) {
 #undef PARSE_UNSIGNED_ITERATION
 #undef PARSE_UNSIGNED_ITERATION_LAST
 
+
+
+template <typename T>
+bool ParseHex(const char* s, size_t length, T* out) {
+  T result = 0;
+  int num_iterations = (int)(sizeof(T)*2);
```

Comment:
Since we already know that `length` will be `<= sizeof(T)*2`, it does not seem useful to have this. Just loop on `length`.

Reply:
good point. Implemented this